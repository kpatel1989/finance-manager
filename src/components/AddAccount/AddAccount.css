.add-account {
    max-width: 500px!important;
    border: 1px solid #CACACA;
    margin: 10px;
}