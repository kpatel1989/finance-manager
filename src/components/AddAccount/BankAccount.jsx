import React, { useState } from 'react';
import { Col, Form, Row } from 'react-bootstrap';

function BankAccount() {
  const [bankName, setBankName] = useState();
  const [currentAmount, setCurrentAmount] = useState();
  const [accountNumber, setAccountNumber] = useState();
  return (
    <>
      <Form.Group as={Row} controlId="bankName">
        <Form.Label column sm={4}>Bank Name</Form.Label>
        <Col sm={8}>
          <Form.Control type="text" placeholder="Enter bank name" value={bankName} onChange={(e) => setBankName(e.target.value)} />
        </Col>
      </Form.Group>
      <Form.Group as={Row} controlId="accountNumber">
        <Form.Label column sm={4}>Account number</Form.Label>
        <Col sm={8}>
          <Form.Control type="text" placeholder="Enter the last 5 digits of this account" value={accountNumber} onChange={(e) => setAccountNumber(e.target.value)} />
        </Col>
      </Form.Group>
      <Form.Group as={Row} controlId="balance">
        <Form.Label column sm={4}>Current amount</Form.Label>
        <Col sm={8}>
          <Form.Control type="text" placeholder="Enter the current balance of this account" value={currentAmount} onChange={(e) => setCurrentAmount(e.target.value)} />
        </Col>
      </Form.Group>
    </>
  );
}

export default BankAccount;
