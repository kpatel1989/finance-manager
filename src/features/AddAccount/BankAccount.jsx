import React from 'react';
import { Col, Form, Row } from 'react-bootstrap';

function BankAccount(props) {
  return (
    <>
      <Form.Group as={Row} controlId="bankName">
        <Form.Label column sm={4}>Bank Name</Form.Label>
        <Col sm={8}>
          <Form.Control type="text" placeholder="Enter bank name" value={props.bankName} onChange={(e) => props.setBankName(e.target.value)} />
        </Col>
      </Form.Group>
      <Form.Group as={Row} controlId="accountNumber">
        <Form.Label column sm={4}>Account number</Form.Label>
        <Col sm={8}>
          <Form.Control type="text" placeholder="Enter the last 5 digits of this account" value={props.accountNumber} onChange={(e) => props.setAccountNumber(e.target.value)} />
        </Col>
      </Form.Group>
      <Form.Group as={Row} controlId="balance">
        <Form.Label column sm={4}>Current amount</Form.Label>
        <Col sm={8}>
          <Form.Control type="text" placeholder="Enter the current balance of this account" value={props.currentAmount} onChange={(e) => props.setCurrentAmount(e.target.value)} />
        </Col>
      </Form.Group>
    </>
  );
}

export default BankAccount;
