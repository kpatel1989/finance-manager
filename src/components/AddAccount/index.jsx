import React, { useState } from 'react';
import {
  Button, Card, Col, Form, Row,
} from 'react-bootstrap';
import './AddAccount.css';
import BankAccount from './BankAccount';

const ACCOUNT_TYPES = {
  chequing: 'Chequing',
  savings: 'Savings',
  creditCard: 'Credit Card',
  rrsp: 'RRSP',
  tfsa: 'TFSA',
};

function renderFormForAccountType(accountType) {
  switch (accountType) {
    case ACCOUNT_TYPES.chequing:
      return <BankAccount />;
    case ACCOUNT_TYPES.savings:
      return <BankAccount />;
    default:
      return '';
  }
}

function AddAccount() {
  const [accountName, setAccountName] = useState();
  const [accountType, setAccountType] = useState();
  return (
    <Card>
      <Card.Body>
        <Card.Title>Add a new account</Card.Title>
        <Form align="left">
          <Form.Group as={Row} controlId="accountName">
            <Form.Label column sm={4}>Account Name</Form.Label>
            <Col sm={8}>
              <Form.Control type="text" placeholder="Enter account name" value={accountName} onChange={(e) => setAccountName(e.target.value)} />
            </Col>
          </Form.Group>
          <Form.Group as={Row} controlId="accountType">
            <Form.Label column sm={4}>Account Type</Form.Label>
            <Col sm={8}>
              <Form.Control as="select" custom onChange={(e) => setAccountType(e.target.value)} value={accountType}>
                <option disabled selected>Select the type of account</option>
                <option>Chequing</option>
                <option>Savings</option>
                <option>Credit Card</option>
                <option>Pension</option>
                <option>TFSA</option>
              </Form.Control>
            </Col>
          </Form.Group>
          {renderFormForAccountType(accountType)}
        </Form>
        <Button variant="primary">Save</Button>
      </Card.Body>
    </Card>
  );
}

export default AddAccount;
