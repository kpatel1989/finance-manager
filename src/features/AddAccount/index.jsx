import React, { useState } from 'react';
import {
  Button, Card, Col, Form, Row,
} from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import { addAccount } from '../../store/reducers/Account';
import './AddAccount.css';
import BankAccount from './BankAccount';

const ACCOUNT_TYPES = {
  chequing: 'Chequing',
  savings: 'Savings',
  creditCard: 'Credit Card',
  rrsp: 'RRSP',
  tfsa: 'TFSA',
};

export default function AddAccount() {
  const [accountName, setAccountName] = useState();
  const [accountType, setAccountType] = useState();
  const [bankName, setBankName] = useState();
  const [accountNumber, setAccountNumber] = useState();
  const [currentAmount, setCurrentAmount] = useState();

  const dispatch = useDispatch();
  const saveAccount = () => {
    dispatch(addAccount({
      accountName,
      accountType,
      bankName,
      accountNumber,
      currentAmount
    }));
  }
  const renderFormForAccountType = (accountType) => {
    switch (accountType) {
      case ACCOUNT_TYPES.chequing:
      case ACCOUNT_TYPES.savings:

        return <BankAccount
          bankName={bankName} setBankName={setBankName}
          accountNumber={accountNumber} setAccountNumber={setAccountNumber}
          currentAmount={currentAmount} setCurrentAmount={setCurrentAmount}
        />;
      default:
        return '';
    }
  }
  return (
    <Card>
      <Card.Body>
        <Card.Title>Add a new account</Card.Title>
        <Form align="left">
          <Form.Group as={Row} controlId="accountName">
            <Form.Label column sm={4}>Account Name</Form.Label>
            <Col sm={8}>
              <Form.Control type="text" placeholder="Enter account name" value={accountName} onChange={(e) => setAccountName(e.target.value)} />
            </Col>
          </Form.Group>
          <Form.Group as={Row} controlId="accountType">
            <Form.Label column sm={4}>Account Type</Form.Label>
            <Col sm={8}>
              <Form.Control as="select" custom onChange={(e) => setAccountType(e.target.value)} value={accountType}>
                <option disabled>Select the type of account</option>
                <option>Chequing</option>
                <option>Savings</option>
                <option>Credit Card</option>
                <option>Pension</option>
                <option>TFSA</option>
              </Form.Control>
            </Col>
          </Form.Group>
          {renderFormForAccountType(accountType)}
          <Button variant="primary" onClick={() => saveAccount()}>Save</Button>
        </Form>
      </Card.Body>
    </Card>
  );
}

