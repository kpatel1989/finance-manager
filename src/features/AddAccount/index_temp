import React from 'react';
import {
  Button, Card, Col, Form, Row,
} from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import { addAccount } from '../../store/reducers/Account';
import './AddAccount.css';
import BankAccount from './BankAccount';

const ACCOUNT_TYPES = {
  chequing: 'Chequing',
  savings: 'Savings',
  creditCard: 'Credit Card',
  rrsp: 'RRSP',
  tfsa: 'TFSA',
};

export default class AddAccount extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      accountName: '',
      accountType: '',
      bankName: '',
      accountNumber: '',
      currentAmount: ''
    }
    this.dispatch = useDispatch();
  }
  saveAccount() {
    console.log(this.state);
    this.dispatch(addAccount({
      accountName: this.state.accountName,
      accountType: this.state.accountType,
    }));
  }
  renderFormForAccountType(accountType) {
    switch (accountType) {
      case ACCOUNT_TYPES.chequing:
      case ACCOUNT_TYPES.savings:

        return <BankAccount
          bankName={this.state.bankName}
          accountNumber={this.state.accountNumber}
          currentAmount={this.state.currentAmount}
          onChange={(update) => this.setState(update)}
        />;
      default:
        return '';
    }
  }
  render() {
    return (
      <Card>
        <Card.Body>
          <Card.Title>Add a new account</Card.Title>
          <Form align="left">
            <Form.Group as={Row} controlId="accountName">
              <Form.Label column sm={4}>Account Name</Form.Label>
              <Col sm={8}>
                <Form.Control type="text" placeholder="Enter account name" value={this.state.accountName} onChange={(e) => this.setState({ accountName: e.target.value })} />
              </Col>
            </Form.Group>
            <Form.Group as={Row} controlId="accountType">
              <Form.Label column sm={4}>Account Type</Form.Label>
              <Col sm={8}>
                <Form.Control as="select" custom onChange={(e) => this.setState({ accountType: e.target.value })} value={this.state.accountType}>
                  <option disabled>Select the type of account</option>
                  <option>Chequing</option>
                  <option>Savings</option>
                  <option>Credit Card</option>
                  <option>Pension</option>
                  <option>TFSA</option>
                </Form.Control>
              </Col>
            </Form.Group>
            {this.renderFormForAccountType(this.state.accountType)}
          <Button variant="primary" onClick={() => this.saveAccount()}>Save</Button>
          </Form>
        </Card.Body>
      </Card>
    );
  }
}

