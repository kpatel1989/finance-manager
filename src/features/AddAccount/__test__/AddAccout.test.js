import React from 'react';
import { render, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer';

// import { render, fireEvent, cleanup } from 'test-utils';
import AddAccount from '../index';

jest.mock('react-redux');
jest.mock('../../../store/reducers/Account');


afterEach(cleanup);
test('renders the Add Account component', () => {
    const component = renderer.create(<AddAccount />, document.body);
    expect(component.toJSON()).toMatchSnapshot();
});