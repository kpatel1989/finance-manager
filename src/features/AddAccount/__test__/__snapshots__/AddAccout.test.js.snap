// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Add Account component 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Add a new account
    </div>
    <form
      align="left"
      className=""
    >
      <div
        className="form-group row"
      >
        <label
          className="form-label col-form-label col-sm-4"
          htmlFor="accountName"
        >
          Account Name
        </label>
        <div
          className="col-sm-8"
        >
          <input
            className="form-control"
            id="accountName"
            onChange={[Function]}
            placeholder="Enter account name"
            type="text"
          />
        </div>
      </div>
      <div
        className="form-group row"
      >
        <label
          className="form-label col-form-label col-sm-4"
          htmlFor="accountType"
        >
          Account Type
        </label>
        <div
          className="col-sm-8"
        >
          <select
            className="custom-select"
            id="accountType"
            onChange={[Function]}
          >
            <option>
              Select the type of account
            </option>
            <option>
              Chequing
            </option>
            <option>
              Savings
            </option>
            <option>
              Credit Card
            </option>
            <option>
              Pension
            </option>
            <option>
              TFSA
            </option>
          </select>
        </div>
      </div>
      
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Save
      </button>
    </form>
  </div>
</div>
`;
